common/workunit/workunit.cpp

```diff
@@ -9476,6 +9923,103 @@ IWUGraphStats *CLocalWorkUnit::updateStats(const char *graphName, StatisticCreat
     return new CWuGraphStats(LINK(p), creatorType, creator, _wfid, graphName, subgraph);
 }
 
+bool CLocalWorkUnit::usingDedicatedLogFiles(const char *processType, const char *processName, const char *logSpec) const
+{
+    if (isEmptyString(processType) || (isEmptyString(processName) && isEmptyString(logSpec)))
+        return false;
+
+    IArrayOf<IConstWUProcessLogInfo> processLogs;
+    if (!isEmptyString(processName))
+        getWUProcessLogInfo(processType, processName, processLogs);
+    else
+    {
+        //ECLWatch may retrieve thor master log using the logSpec without specifying processName.
+        //We need to parse the processName from the logSpec.
+        StringBuffer path, logName, process;
+        splitFilename(logSpec, NULL, &path, &logName, NULL);
+
+        const char *searchStr = strieq(processType, "Thor") ? THORMASTERLOGSEARCHSTR : ECLAGENTLOGSEARCHSTR;
+        if (strncmp(logName, searchStr, strlen(searchStr)) != 0) //example: mythor_thormaster 
+        {// This is an imported WU. Find processName from logName.
+            VStringBuffer afterStr("_%s", searchStr);
+            const char *ptr = logName.str();
+            const char *pptr = strstr(ptr, afterStr);
+            if (!pptr)
+                return false; //Should not happen
+            process.append(pptr - ptr, ptr);
+        }
+        else if (!path.isEmpty())
+        {// Find processName from path
+            path.setLength(path.length() - 1); //Remove the last char which is a PATHSEPCHAR
+            const char *ptr = strrchr(path.str(), PATHSEPCHAR);
+            if (!ptr)
+                return false; //Should not happen
+            process.set(ptr + 1);
+        }
+        else
+            return false; //Should not happen
+
+        getWUProcessLogInfo(processType, process, processLogs);
+    }
+    if (processLogs.ordinality() == 0)
+        return false;
+
+    return processLogs.item(0).getSingleLog();
```

Comment:
I think similar to previous comment, that processLogs.ordinality() should never be >1.
Should assert that's true and add comment though.

Reply:
In fact, if the WU is not imported and run overnight, the processLogs.ordinality() will > 1. Could the SingleLog flags be different from one processLog to another processLog?